### DISCUSSION TITLE
Improve dark mode: adapt to system + make it configurable in ApiReference configuration

### DISCUSSION THREAD

**Discussion by marcklingen** [scalar/scalar#728] [Ideas]
I embed the scalar api reference into our documentation page. The docs page already manages dark mode as it adapts to the system and has a toggle to manually change from light to dark.

I'd suggest two improvements
- scalar should adapt to the system
- it would be nice to change the dark mode state via the ApiReference configuration as our page already manages dark mode state and I'd like to pass it to ApiReference

Demo, initially changing OS system appearance, then using our docs page's toggle
![CleanShot 2023-12-24 at 13 19 19](https://github.com/scalar/scalar/assets/2834609/0c705b93-5443-4e8b-8fbf-336a2f53afeb)

**Comment by marclave** [comment]
@marcklingen great idea! I think the best first step would be to make it configurable which allows the most flexibility :)

**Reply by marcklingen** [reply]
Agree on configuring it. Suggestion:

- Default to scalar theme switch
- set theme via var: dark/light
- alternatively: system

Alternatively, change default logic to be similar as Nextra.site (system as default), optionally switch to light/dark

**Reply by marclave** [reply]
oh wow i wish i saw this before i started coding!

https://github.com/scalar/scalar/pull/729

i left a comment about how we should also offer the ability to hide the toggle button in the case where docs sites have it already. feel free to add more comments in the pr :)
